src/main/java/com/spotifyapi/service/impl/SpotifyTrackServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.spotifyapi.service.impl;
+
+import com.spotifyapi.model.SpotifyTrackFromPlaylist;
+import com.spotifyapi.model.SpotifyUserPlaylist;
+import com.spotifyapi.repository.PlaylistRepository;
+import com.spotifyapi.repository.TrackRepository;
+import com.spotifyapi.service.SpotifyTrackService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import se.michaelthelin.spotify.model_objects.specification.Track;
+import se.michaelthelin.spotify.model_objects.specification.TrackSimplified;
+
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class SpotifyTrackServiceImpl implements SpotifyTrackService {
+
+    private final TrackRepository trackRepository;
+    private final PlaylistRepository playlistRepository;
+
+    @Override
+    public void saveTracks(Track track, SpotifyUserPlaylist playlist) {
+        SpotifyTrackFromPlaylist trackFromPlaylist = new SpotifyTrackFromPlaylist();
+        Optional<SpotifyUserPlaylist> spotifyUserPlaylist = playlistRepository.findById(playlist.getId());
+
+
+        if(!isAlreadyExist(track.getId(), spotifyUserPlaylist.get().getId())) {
+            trackFromPlaylist.setId(track.getId());
+            trackFromPlaylist.setName(track.getName());
+            trackFromPlaylist.setExternalUrl(track.getExternalUrls().get("spotify"));
+            trackFromPlaylist.setArtistName(track.getArtists()[0].getName());
```

Comment:
NPE?

Reply:
Resolved problem with help to check track is null